### DISCUSSION TITLE
Very strange/low quality fuzzy matching behavior

### DISCUSSION THREAD

**Discussion by diego898** [junegunn/fzf#3791] [Q&A] [answered]
For the past ~1 year Ive noticed a "decrease" in the quality of the "'fuzzyness" matching - I cant really determine if this is my fault (some setting), etc. 

Here is what I mean - when I search my command history:

If I type sudo - I get this set of matches, with the one I want selected
![image](https://github.com/junegunn/fzf/assets/742407/a6f57e1a-6304-4d73-9e34-419245d21855)

If i make a single/simple typo:
![image](https://github.com/junegunn/fzf/assets/742407/014f4654-aedb-476d-8fa3-a745c800d8f9)

not at all what I expect...?

Any help would be appreciated!

**Comment by LangLangBart** [comment, marked as answer]
I can reproduce the same issue as you outlined in your report for `sudo/wsudo`.

It shows better results when the order of characters is correct, e.g., `tcdump`.

<img src="https://github.com/junegunn/fzf/assets/92653266/dd6fae30-1047-4481-adc5-158f448caa70" width="600">

---

**Related Comment by the Maintainer:** 
- [FZF feels less tolerant of typos (gif example) · Issue #3648 · junegunn/fzf · GitHub](https://github.com/junegunn/fzf/issues/3648#issuecomment-1970134409)

**Reply by diego898** [reply]
Thanks for your reply! This is unfortunate! I make typos (which I guess means out of order?) frequently.

**Reply by diego898** [reply]
@LangLangBart - do you know of any way to make this… fuzzier? That is… typo tolerant? Perhaps ordering by most frequently used commands?

**Reply by junegunn** [reply]
No, for that to work (allowing actual typos, not omissions), fzf needs a completely different matching/scoring algorithm. fzf is heavily optimized around the current algorithm and it's unlikely that we will add a different one.
